### PR TITLE
feat!: remove compat variables for previous compat stylesheets

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**
[AB#3023827](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/3023827)

BREAKING CHANGE: Custom properties used by compatibility stylesheets for previous SKY UX major versions have been removed. Consumers should adopt the updated styles within their applications.

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2598/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/a11y-storybook/)
- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/colorpicker-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/core-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/errors-storybook/)
- [flyout-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/flyout-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/forms-storybook/)
- [help-inline-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/help-inline-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/indicators-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/lookup-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/modals-storybook/)
- [navbar-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/navbar-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/phone-field-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/popovers-storybook/)
- [progress-indicator-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/progress-indicator-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/split-view-storybook/)
- [storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/tiles-storybook/)
- [toast-storybook](https://blackbaud.github.io/skyux-pr-preview/2598/storybooks/toast-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2598/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2598/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2598/playground/)

File: libs/components/layout/src/lib/modules/action-button/action-button-container.default.component.scss [hunk 1]
```diff
@@ -30,10 +30,7 @@
   .sky-action-button-flex {
     display: block;
     padding: 0;
-    margin: calc(
-        var(--sky-compat-action-button-flex-margin, $sky-margin-double) * -1
-      )
-      0;
+    margin: -$sky-margin-double 0;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/layout/src/lib/modules/action-button/action-button-container.default.component.scss]: SCSS recommends wrapping unary operators in parenthesis (e.g. `(-$sky-margin-double)`).
See: https://sass-lang.com/documentation/operators/numeric/#unary-operators

File: libs/components/layout/src/lib/modules/description-list/description-list.component.scss [hunk 2]
```diff
@@ -38,8 +38,7 @@
   &.sky-description-list-horizontal-mode {
     display: flex;
     flex-wrap: wrap;
-    margin: 0 0 var(--sky-compat-description-list-margin-bottom, -$sky-space-lg)
-      0;
+    margin: 0 0 (-$sky-space-lg) 0;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/layout/src/lib/modules/description-list/description-list.component.scss:41]: Good use of parenthesis.

File: libs/components/layout/src/lib/modules/action-button/action-button-container.default.component.scss [hunk 3]
```diff
@@ -50,9 +47,8 @@
   .sky-action-button-flex {
     display: flex;
     flex-flow: row wrap;
-    padding: var(--sky-compat-action-button-flex-sm-padding, 0) 0;
-    margin: calc(var(--sky-compat-action-button-flex-margin, $sky-margin) * -1)
-      0;
+    padding: 0;
+    margin: -$sky-margin 0;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/layout/src/lib/modules/action-button/action-button-container.default.component.scss]: Ditto.

**Review by Blackbaud-SteveBrush — COMMENTED** [review]
Any visual test failures with this change?

File: libs/components/layout/src/lib/modules/action-button/action-button-container.default.component.scss (hunk 1 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/layout/src/lib/modules/action-button/action-button-container.default.component.scss]: Done

File: libs/components/layout/src/lib/modules/action-button/action-button-container.default.component.scss (hunk 3 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/layout/src/lib/modules/action-button/action-button-container.default.component.scss]: Done

**Comment by Blackbaud-TrevorBurch** [comment]
> Any visual test failures with this change?

Nope. I had a couple initially but fixed them. The only failures were the expected fluke ones.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**